sjsonnet/src/sjsonnet/PrettyYamlRenderer.scala

```diff
@@ -0,0 +1,434 @@
+package sjsonnet
+
+import java.io.{StringWriter, Writer}
+import java.util.regex.Pattern
+
+import ujson.BaseRenderer
+import upickle.core.{ArrVisitor, ObjVisitor}
+
+/**
+ * A version of YamlRenderer that tries its best to make the output YAML as
+ * pretty as possible: unquoted strings, de-dented lists, etc. Follows the PyYAML
+ * style. Also adds the ability to stream writes to a generic `Writer`.
+ */
+class PrettyYamlRenderer(out: Writer = new java.io.StringWriter(),
+                         indentArrayInObject: Boolean = false,
+                         indent: Int = 2,
+                         idealWidth: Int = 80) extends BaseRenderer[Writer](out, indent){
+  var newlineBuffered = false
+  var dashBuffered = false
+  var afterColon = false
+  var afterKey = false
+  var topLevel = true
+  var leftHandPrefixOffset = 0
+  var firstElementInArray = false
```

Comment:
This variable is used to support rendering the start of nested arrays inline:

```yaml
- - a
  - b
- - c
  - d
```

Rather than the status quo:

```yaml
-
  - a
  - b
-
  - c
  - d
```